[MDS-6329] - Minespace View Permits/Permit Conditions

matbusby-fw opened the pull request:
## Objective 

- Copied various styles from Core to Minespace
- Updated styles where necessary
- Updated banner text for Minespace users
- Updated logic to restrict Minespace users to view only
- Renamed CorePageHeader to CommonPageHeader

[MDS-6329](https://bcmines.atlassian.net/browse/MDS-6329)

<img width="1007" alt="image" src="https://github.com/user-attachments/assets/4df9c4a5-3854-4a26-8c1b-df6bc0c17a47" />

matbusby-fw committed:
Add permit components into Minespace

matbusby-fw committed:
Refactor CorePageHeader to CommonPageHeader and update to be used in both Core and Minespace.  Copied various styles to Minespace and updated permit condition logic to not allow editing options while in Minespace

matbusby-fw committed:
Refactor CorePageHeader to CommonPageHeader and update to be used in both Core and Minespace.  Copied various styles to Minespace and updated permit condition logic to not allow editing options while in Minespace

matbusby-fw committed:
trimmed new button styles to affect less parts of Minespace

matbusby-fw committed:
adjust timezone tests

taraepp commented on services/common/src/components/permits/PermitReviewBanner.tsx:
```diff
@@ -1,55 +1,76 @@
+        }}
+      >
+        <Col>{<FontAwesomeIcon icon={paramsToUse.icon} />}</Col>
+        <Col span={23}>{paramsToUse.text}</Col>
```
Just checking- visually looks the same/better in Core as before?

matbusby-fw replied:
Looks the same for the most part.  Could end up off by a few pixels, but without it the long text was taking up the whole width and leaving the icon on it's own row.

taraepp commented on services/minespace-web/src/components/dashboard/mine/permits/ViewPermitRedirect.tsx:
```diff
@@ -0,0 +1,64 @@
+      permit.permit_amendments[0].permit_amendment_guid,
+      tab
+    );
+    console.log(
```
Great console log testing, I am 100% that the redirect is working. 😂 
But probably should be taken out.

matbusby-fw replied:
Ha, yes.  It is working!  I had a conflict in the route paths that were failing to hit the redirect path at one point, so had to troubleshoot. :grin:

taraepp commented on services/minespace-web/src/constants/routes.ts:
```diff
@@ -301,3 +305,26 @@ export const REPORT_VIEW_EDIT = {
 };
+
+export const VIEW_MINE_PERMIT_AMENDMENT = {
+  route: "/mines/:id/permits/:permitGuid/permit-amendment/:permitAmendmentGuid/:tab",
```
I know it's kind of annoying- but can we use mineGuid instead of id in these new routes? `id` is just so vague that in the context of the page you have to think an extra second about which object it refers to.

matbusby replied:
Yeah, I hear you.  I generally prefer mineGuid as well, but left it as id here to match the usage of the identical route in Core.  But, I guess there's no reason they have to match.

matbusby-fw committed:
Remove console logs and update route parameter naming.

matbusby-fw committed:
Refactor useParams destructuring in ViewPermitRedirect.

Updated destructuring of useParams to align with consistent naming conventions and improve readability. This change has no impact on functionality but enhances code clarity and maintainability.

matbusby-fw committed:
revert change from id to mineGuid in the params to maintain existing functionality

matbusby-fw committed:
added to ViewPermit.spec.tsx to include a test that Minespace users won't see unverified conditions.
